### PR TITLE
tinymist: 0.11.13 -> 0.11.14

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Changelog: https://github.com/Myriad-Dreamin/tinymist/blob/refs/tags/v0.11.14/CHANGELOG.md

- removal of now useless packages `typst-preview` and `vscode-extensions.mgt19937.typst-preview`.

cc @LamprosPitsillos 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 325275` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>tinymist</li>
    <li>vscode-extensions.myriad-dreamin.tinymist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 325275` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>tinymist</li>
    <li>vscode-extensions.myriad-dreamin.tinymist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 325275` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>tinymist</li>
    <li>vscode-extensions.myriad-dreamin.tinymist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 325275` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>tinymist</li>
    <li>vscode-extensions.myriad-dreamin.tinymist</li>
  </ul>
</details>

**Comment by drupol** [comment]
There's a big change in that release.

`typst-preview` should be disabled because it has been integrated in this extension: https://github.com/Myriad-Dreamin/tinymist/blob/refs/tags/v0.11.14/editors/vscode/CHANGELOG.md

See the deprecation in the repo: https://github.com/Enter-tainer/typst-preview

**Comment by drupol** [comment]
I think the `typst-preview` vscode extension should be removed too.

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 1]
```diff
@@ -3040,7 +3040,9 @@ let
         };
       };
 
-      mgt19937.typst-preview = callPackage ./mgt19937.typst-preview { };
+      mgt19937.typst-preview = throw ''
```
Comment (drupol) [inline pkgs/applications/editors/vscode/extensions/default.nix:3043]: Can we add `throw` in here?

Comment (GaetanLepage) [reply pkgs/applications/editors/vscode/extensions/default.nix:3043]: Apparently yes.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 325275` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>tinymist</li>
    <li>vscode-extensions.myriad-dreamin.tinymist</li>
  </ul>
</details>

File: pkgs/applications/editors/vscode/extensions/default.nix (hunk 1 above)
Comment (drupol) [reply pkgs/applications/editors/vscode/extensions/default.nix:3043]: Maybe it should be moved to the aliases section ?

Comment (GaetanLepage) [reply pkgs/applications/editors/vscode/extensions/default.nix]: Sure ! Done.

**Review by drupol — APPROVED** [review]
LGTM now, thanks :)

**Comment by drupol** [comment]
@ofborg build vscode-extensions.myriad-dreamin.tinymist

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -1372,6 +1372,7 @@ mapAliases ({
   trustedGrub-for-HP = throw "trustedGrub-for-HP has been removed, because it is not maintained upstream anymore"; # Added 2023-05-10
   tvbrowser-bin = tvbrowser; # Added 2023-03-02
   typst-fmt = typstfmt; # Added 2023-07-15
+  typst-preview = throw "The features of this program have been consolidated to 'tinymist', an all-in-one language server for typst"; # Added 2024-07-07
```
Comment (inmaldrerah) [inline pkgs/top-level/aliases.nix:1375]: This seems quite unclear: the user will only be prompted with just the message thrown, saying "this program" is abandoned, having no clues about which program "this program" actually is.

Comment (GaetanLepage) [reply pkgs/top-level/aliases.nix:1375]: On my side, running `nix build nixpkgs#typst-preview` shows the correct message:

```
❯ nix build nixpkgs#typst-preview
error:
       … while evaluating a branch condition

         at /nix/store/yzkx7gx2cn73dh3c7zg5rvgkp3yli1cd-source/pkgs/top-level/aliases.nix:33:5:

           32|   removeDistribute = alias: with lib;
           33|     if isDerivation alias then
             |     ^
           34|       dontDistribute alias

       … while evaluating a branch condition

         at /nix/store/yzkx7gx2cn73dh3c7zg5rvgkp3yli1cd-source/pkgs/top-level/aliases.nix:26:5:

           25|   removeRecurseForDerivations = alias: with lib;
           26|     if alias.recurseForDerivations or false
             |     ^
           27|     then removeAttrs alias [ "recurseForDerivations" ]

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: The features of this program have been consolidated to 'tinymist', an all-in-one language server for typst
```

Comment (wlib) [reply pkgs/top-level/aliases.nix:1375]: @GaetanLepage I had to full text search this repo on github to find out it was for `typst-preview`, so it's probably best to make it "The features of ~this program~typst-preview have been consolidated..."

Comment (GaetanLepage) [reply pkgs/top-level/aliases.nix:1375]: Done !

**Pull request merged**